block/src/qcow/mod.rs

```diff
@@ -439,7 +439,7 @@ pub struct QcowFile {
 
 impl QcowFile {
     /// Creates a QcowFile from `file`. File must be a valid qcow2 image.
-    pub fn from(mut file: RawFile) -> Result<QcowFile> {
+    pub fn from(mut file: RawFile, max_nesting_depth: u32) -> Result<QcowFile> {
```

Comment:
Can you keep the prototype of `::from()` the same and have a method like `::from_with_nesting_depth()` which you call from that with the constant. This will mean this change will be a lot less intrusive and only the qcow code will need to know the limit. Obviously need to make the change below too...

Reply:
Yes, this is a very useful suggestion. I've updated it.